PencilCase/PencilCase/scripts/Pages/ImportProduct.ts

```diff
@@ -1,143 +1,32 @@
-﻿import { PageBase } from './PageBase';
-import { Navigator } from '../Navigator';
+﻿import { DealPageBase } from './DealPageBase';
 import * as Consts from './Consts';
-import { Product } from '../Models/Product';
-import { ProductRepository } from '../Repositories/ProductRepository';
-import { OrderRepository } from '../Repositories/OrderRepository';
 import { Order } from '../Models/Order';
 import * as Utils from '../Utils';
 import { OrderTypes } from '../Models/Order';
 
-export class ImportProduct extends PageBase {
-    private navigator: Navigator = Navigator.instance;
-    private products: KnockoutObservableArray<Product> = ko.observableArray([]);
-    private dict = {};
-    private selectOptions = { Id: "placeholder", Name: "选择产品……", WholesalePrice: 0, WholesaleUnitName: '' };
-    private selectedProductId: KnockoutObservable<string> = ko.observable(this.selectOptions.Id);
-    private selectedProduct: KnockoutObservable<any> = ko.observable(this.selectOptions);
-    private productSelected = ko.computed(() => {
-        if (this.selectedProduct() && this.selectedProduct().Id !== this.selectOptions.Id)
-            return true;
-        return false;
-    });
-
-    private selectedProductQuantity: KnockoutObservable<number> = ko.observable(1);
+export class ImportProduct extends DealPageBase {
     private selectedProductImportPrice: KnockoutObservable<number> = ko.observable(0);
-    private clearSelection = ko.observable(false);
-
-    private importOrders = ko.observableArray<Order>([]);
-    private batchId: string = null;
-    private totalNumber: KnockoutObservable<number> = ko.observable(0);
-    private totalPrice: KnockoutObservable<number> = ko.observable(0);
-
-    private productRepository = new ProductRepository();
-    private orderRepository = new OrderRepository();
 
     constructor() {
         super();
         this.title = ko.observable("进货");
         this.pageId = Consts.Pages.ImportProduct.Id;
-        this.back = Navigator.instance.goHome;
-        this.selectedProductId.subscribe((newValue: string) => {
+        this.onSelectionChanged = (newValue: string) => {
```

Comment:
Because on selection changed handler is different between (Retail, Wholesale) and Import, so couldn't put this part into the base class, or it won't work.